cli/src/pcluster/resources/compute_node/user_data.sh

```diff
@@ -151,6 +151,16 @@ write_files:
         done;
         export HOME="${!HOME_BAK}"
       }
+
+      function measure_startup_time
+      {
+        METRIC_NAMESPACE="ParallelCluster"
+        METRIC_NAME="StartupTime"
+        INSTANCE_ID=$(cat /etc/parallelcluster/slurm_plugin/slurm_node_spec.json | jq -r .instance_id)
+        INSTANCE_TYPE=$(cat /etc/parallelcluster/slurm_plugin/slurm_node_spec.json | jq -r .'compute."instance-type"')
+        aws cloudwatch put-metric-data --namespace $METRIC_NAMESPACE --metric-name $METRIC_NAME --value $(($2-$1)) --dimensions InstanceID=$INSTANCE_ID,InstanceType=$INSTANCE_TYPE --region "${AWS::Region}"
```

Comment:
(1) I suggest use variables like `start`, `end` at the beginning to get the parameters passed in to improve the readability. Otherwise, it's unclear what `$2 - $1` is doing here, unless looking at how the function is called. 
(2) I suggest add an error handling logic for the put-metric-data call, such as `|| echo "[WARNING] failed to push the metric StartupTime to CloudWatch."` at the end?
(3) Do we need to update the integration test to check if this metric is generated? But because of the failure tolerance here, we probably don't want to fail the integration test if the metric wasn't created anyway. @davprat 
(4) If we want to add this metric to CloudWatch Dashboard, how will the data be presented in the graph according to the 2 dimensions defined here?

Reply:
Done (1)(2)

Reply:
Please also respond to the other 2 questions (whatever decision/reasoning)

Reply:
(3) After discussion we decided yes the integration test needs to be updated. We can create another PR.
(4) After discussion we decided leave adding to Cloudwatch Dashboard to the future